code/obj/item.dm

```diff
@@ -446,17 +459,15 @@
 		the_dir = turn(the_dir,45)
 */
 
-/obj/item/proc/combust_ended()
-	processing_items.Remove(src)
-
 /obj/item/proc/combust() // cogwerks- flammable items project
 	if (!src.burning)
 		src.visible_message("<span class='alert'>[src] catches on fire!</span>")
 		src.burning = 1
+		src.firesource = 1
 		if (src.burn_output >= 1000)
-			src.overlays += image('icons/effects/fire.dmi', "2old")
+			UpdateOverlays(image('icons/effects/fire.dmi', "2old"),"burn_overlay")
 		else
-			src.overlays += image('icons/effects/fire.dmi', "1old")
+			UpdateOverlays(image('icons/effects/fire.dmi', "1old"),"burn_overlay")
```

Comment:
Oh, good catch! Having direct `overlays` modification on `item` was bad.

Reply:
eeee this made me happy :cherry_blossom: